### PR TITLE
fix(rekor): enable rekor search ui by default

### PR DESCRIPTION
**Pull request by kahboom**
This PR updates the operator to enable the Rekor Search UI by default.

Changes:
- enable default to true
- swap enabled flag to a pointer rather than a boolean, regenerated types
- updates relevant tests to use the updated type
- update tests that compare rekor instance objects to check specific properties instead

NOTE: I'm experiencing some failures locally for CTLog and the following for Rekor, so leaving this as a draft for now:

```
Summarizing 2 Failures:
  [FAIL] Rekor controller Rekor controller test [It] should successfully reconcile a custom resource for Rekor
  /secure-sign-operator/controllers/rekor/rekor_controller_test.go:219
  [FAIL] Rekor hot update test Rekor update test [It] should successfully reconcile a custom resource for Rekor
  /secure-sign-operator/controllers/rekor/rekor_hot_update_test.go:146

Ran 2 of 2 Specs in 133.229 seconds
FAIL! -- 0 Passed | 2 Failed | 0 Pending | 0 Skipped
```

I've run out of time today so I'll continue to look at it next week. cc @lance @osmman @JasonPowr

### CODE REVIEW

File: controllers/common/utils/optional.go [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package utils
 
+func IsEnabled(flag *bool) bool {
```
Comment (kahboom) [inline controllers/common/utils/optional.go:3]: This function is redundant as it does the same as `OptionalBool` but I felt the name wouldn't have been very clear to future devs and I didn't want to rename `OptionalBool`, however, I'm happy to make any suggested changes here!

File: controllers/rekor/actions/initialize.go [hunk 2]
```diff
@@ -24,15 +25,17 @@ func (i initializeAction) Name() string {
...
+	if c == nil {
+		return false
+	}
+	return (c.Reason == constants.Creating || c.Reason == constants.Ready) && utils.IsEnabled(instance.Spec.RekorSearchUI.Enabled)
 }
 
 func (i initializeAction) Handle(ctx context.Context, instance *rhtasv1alpha1.Rekor) *action.Result {
 
-	if (!instance.Spec.RekorSearchUI.Enabled || meta.IsStatusConditionTrue(instance.Status.Conditions, UICondition)) &&
+	if (instance.Spec.RekorSearchUI.Enabled == nil || !*instance.Spec.RekorSearchUI.Enabled || meta.IsStatusConditionTrue(instance.Status.Conditions, UICondition)) &&
```
Comment (lance) [inline controllers/rekor/actions/initialize.go:36]: Could you use the `IsEnabled()` function here?

**Review by JasonPowr — COMMENTED** [review]
Some of the e2e tests still need to be updated, for example here: https://github.com/securesign/secure-sign-operator/blob/main/e2e/common_install_test.go#L62

and here: https://github.com/securesign/secure-sign-operator/blob/main/e2e/config_update_test.go#L66

File: api/v1alpha1/rekor_types_test.go [hunk 3]
```diff
@@ -188,7 +188,9 @@ var _ = Describe("Rekor", func() {
 					Expect(k8sClient.Create(context.Background(), &rekorInstance)).To(Succeed())
 					fetched := &Rekor{}
 					Expect(k8sClient.Get(context.Background(), getKey(&rekorInstance), fetched)).To(Succeed())
-					Expect(fetched.Spec).To(Equal(expectedRekorInstance.Spec))
+					Expect(fetched.Spec.Pvc.Name).To(Equal(expectedRekorInstance.Spec.Pvc.Name))
+					Expect(fetched.Spec.Pvc.Size).To(Equal(expectedRekorInstance.Spec.Pvc.Size))
+					Expect(fetched.Spec.SearchUI.Enabled).To(BeTrue())
```
Comment (osmman) [inline api/v1alpha1/rekor_types_test.go:193]:
```suggestion
					Expect(fetched.Spec.RekorSearchUI.Enabled).To(BeTrue())
```

File: api/v1alpha1/rekor_types_test.go [hunk 4]
```diff
@@ -270,7 +272,9 @@ var _ = Describe("Rekor", func() {
 					Expect(k8sClient.Create(context.Background(), &rekorInstance)).To(Succeed())
 					fetchedRekor := &Rekor{}
 					Expect(k8sClient.Get(context.Background(), getKey(&rekorInstance), fetchedRekor)).To(Succeed())
-					Expect(fetchedRekor.Spec).To(Equal(expectedRekorInstance.Spec))
+					Expect(fetchedRekor.Spec.Pvc.Name).To(Equal(expectedRekorInstance.Spec.Pvc.Name))
+					Expect(fetchedRekor.Spec.Pvc.Size).To(Equal(expectedRekorInstance.Spec.Pvc.Size))
+					Expect(fetchedRekor.Spec.SearchUI.Enabled).To(BeTrue())
```
Comment (osmman) [inline api/v1alpha1/rekor_types_test.go:277]:
```suggestion
					Expect(fetchedRekor.Spec.RekorSearchUI.Enabled).To(BeTrue())
```

File: api/v1alpha1/rekor_types_test.go [hunk 5]
```diff
@@ -188,7 +188,9 @@ var _ = Describe("Rekor", func() {
 					Expect(k8sClient.Create(context.Background(), &rekorInstance)).To(Succeed())
 					fetched := &Rekor{}
 					Expect(k8sClient.Get(context.Background(), getKey(&rekorInstance), fetched)).To(Succeed())
-					Expect(fetched.Spec).To(Equal(expectedRekorInstance.Spec))
+					Expect(fetched.Spec.Pvc.Name).To(Equal(expectedRekorInstance.Spec.Pvc.Name))
+					Expect(fetched.Spec.Pvc.Size).To(Equal(expectedRekorInstance.Spec.Pvc.Size))
+					Expect(fetched.Spec.RekorSearchUI.Enabled).To(BeTrue())
```
Comment (lance) [inline api/v1alpha1/rekor_types_test.go:193]: Here you need to dereference the boolean pointer. My mistake in the earlier suggestion.
```suggestion
					Expect(*fetched.Spec.RekorSearchUI.Enabled).To(BeTrue())
```

File: api/v1alpha1/rekor_types_test.go [hunk 6]
```diff
@@ -270,7 +272,9 @@ var _ = Describe("Rekor", func() {
 					Expect(k8sClient.Create(context.Background(), &rekorInstance)).To(Succeed())
 					fetchedRekor := &Rekor{}
 					Expect(k8sClient.Get(context.Background(), getKey(&rekorInstance), fetchedRekor)).To(Succeed())
-					Expect(fetchedRekor.Spec).To(Equal(expectedRekorInstance.Spec))
+					Expect(fetchedRekor.Spec.Pvc.Name).To(Equal(expectedRekorInstance.Spec.Pvc.Name))
+					Expect(fetchedRekor.Spec.Pvc.Size).To(Equal(expectedRekorInstance.Spec.Pvc.Size))
+					Expect(fetchedRekor.Spec.RekorSearchUI.Enabled).To(BeTrue())
```
Comment (lance) [inline api/v1alpha1/rekor_types_test.go:277]: Same here - again my earlier example omitted the deferencing.
```suggestion
					Expect(*fetchedRekor.Spec.RekorSearchUI.Enabled).To(BeTrue())
```

File: api/v1alpha1/rekor_types_test.go (hunk 5 above)
Comment (JasonPowr) [inline api/v1alpha1/rekor_types_test.go:193]: I don't think these need to be here, similar to my previous comments the check here is to match a rekor object to a rekor object.

Best to revert these changes

File: api/v1alpha1/rekor_types_test.go (hunk 6 above)
Comment (JasonPowr) [inline api/v1alpha1/rekor_types_test.go:277]: As above

File: controllers/rekor/actions/initialize.go (hunk 2 above)
Comment (JasonPowr) [inline controllers/rekor/actions/initialize.go:36]:
```suggestion
if (!utils.IsEnabled(instance.Spec.RekorSearchUI.Enabled) || meta.IsStatusConditionTrue(instance.Status.Conditions, UICondition)) &&
```

File: api/v1alpha1/rekor_types_test.go (hunk 5 above)
Comment (lance) [reply api/v1alpha1/rekor_types_test.go:193]: But isn't the check meant to ensure that the fetched data is the same as what is expected? If that's the case, you can't compare `fetched` to `expectedRekorInstance` directly because the boolean pointer. You have to compare each field, to determine equivalence don't you?

Comment (JasonPowr) [reply api/v1alpha1/rekor_types_test.go]: Ahhh Ok apologies

Comment (kahboom) [reply api/v1alpha1/rekor_types_test.go]: Thanks for the feedback. Yep @lance this was one of the goals was to avoid comparing directly (even if you could tbh, I prefer not to as it makes tests more fragile, where it makes sense of course).
